src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -140,6 +141,27 @@ public function lightListAction(ProductFilters $filters, Request $request): Resp
         ]);
     }
 
+    /**
+     * The redirection URL is generation thanks to the ProductPreviewProvider however it can't be used in the grid
+     * since the LinkRowAction expects a symfony route, so this action is merely used as a proxy for symfony routing
```

Comment:
have you considered adding an option in the LinkRowAction to toggle between route and direct url?

Reply:
Yeah but the issue is that the rows actions/urls need to be generated differently on each row so that it uses the row's ID For now it's easily done in twig thanks to the `path` twig function and just by adapting the route parameters

But here it would be more complicated as we would need to be able to use the `ProductPreviewProvider` service from twig, by creating a dedicated filter/function I guess, and then it should be configurable to use either router or alternative generation function, it's not impossible but this solution seemed more simple 😅